test/module/irohad/ametsuchi/block_query_test.cpp

```diff
@@ -212,3 +215,154 @@ TEST_F(BlockQueryTest, GetTransactionsWithInvalidTxAndValidTx) {
   });
   ASSERT_TRUE(wrapper.validate());
 }
+
+/**
+ * @given block store with 2 blocks totally containing 3 txs created by
+ * user1@test AND 1 tx created by user2@test
+ * @when get non-existent 1000 block
+ * @then nothing is returned
+ */
+TEST_F(BlockQueryTest, BlockQuery_GetBlocks_GetNonExistentBlock) {
+  auto wrapper = make_test_subscriber<CallExact>(
+      blocks->getBlocks(1000, 1), 0);
+  wrapper.subscribe();
+  ASSERT_TRUE(wrapper.validate());
+}
+
+/**
+ * @given block store with 2 blocks totally containing 3 txs created by
+ * user1@test AND 1 tx created by user2@test
+ * @when height=1, count=1
+ * @then returned exactly 1 block
+ *
+ * @note test for bug, when <total_blocks=2, height=1, count=1>, then
+ * min(1+1, 2)=2, and it reads from 1 to 2 (2 blocks).
+ */
+TEST_F(BlockQueryTest, BlockQuery_GetBlocks_GetExactlyOneBlock) {
+  auto wrapper = make_test_subscriber<CallExact>(
+      blocks->getBlocks(1, 1), 1);
+  wrapper.subscribe();
+  ASSERT_TRUE(wrapper.validate());
+}
+
+/**
+ * @given block store with 2 blocks totally containing 3 txs created by
+ * user1@test AND 1 tx created by user2@test
+ * @when count=0
+ * @then returned empty block
```

Comment:
"empty block" is very confusing. Better to write "return nothing" or similar.

Reply:
Agree, but empty block != nothing
maybe it's better "@then doesn't find such block"?